src/main/java/com/dtolabs/rundeck/plugin/overthere/OTWinRMFileCopier.java

```diff
@@ -0,0 +1,805 @@
+            .mapping(CONFIG_PROTOCOL, PROJ_PROP_PREFIX + WINRM_PROTOCOL)
+            .mapping(CONFIG_CERT_TRUST, PROJ_PROP_PREFIX + WINRM_CERT_TRUST)
+            .mapping(CONFIG_HOSTNAME_VERIFY, PROJ_PROP_PREFIX + WINRM_HOSTNAME_TRUST)
+            .mapping(CONFIG_SPN_ADD_PORT, PROJ_PROP_PREFIX + WINRM_SPN_ADD_PORT)
+            .mapping(CONFIG_SPN_USE_HTTP, PROJ_PROP_PREFIX + WINRM_SPN_USE_HTTP)
+            .mapping(CONFIG_LOCALE, PROJ_PROP_PREFIX + WINRM_LOCALE)
+            .mapping(CONFIG_WINRM_TIMEOUT, PROJ_PROP_PREFIX + WINRM_TIMEOUT)
+            .mapping(CONFIG_TIMEOUT, PROJ_PROP_PREFIX + WINRM_CONNECTION_TIMEOUT_PROPERTY)
+            .mapping(CONFIG_PASSWORD_STORAGE_PATH, PROJ_PROP_PREFIX + WINRM_PASSWORD_STORAGE_PATH)
+            .mapping(CONFIG_DST_FILE, PROJ_PROP_PREFIX + FILE_COPY_DESTINATION_DIR)
+            .build();
+
+    public Description getDescription() {
+        return DESC;
+    }
+
+    @Override
+    public String copyFileStream(ExecutionContext context, InputStream inputStream, INodeEntry node, String destination) throws FileCopierException {
+        ConnectionOptions options = null;
+        String logprompt = "[" + SERVICE_PROVIDER_TYPE + ":" + node.extractHostname() + "] ";
+
+        if (null == context.getExecutionListener()) {
+            System.out.println(logprompt + " Bad plugin context!  NULL ExecutionListener");
+        }
+
+        context.getExecutionListener().log(Constants.VERBOSE_LEVEL, logprompt);
+
+        try {
+            options = willUseConnectionOptions(new ConnectionOptionsBuilder(context, node, framework).build());
+        } catch (ConfigurationException e) {
+            context.getExecutionListener().log(Constants.ERR_LEVEL, logprompt + e.getMessage());
+            throw new FileCopierException(e.getMessage(), StepFailureReason.ConfigurationFailure);
+        }
+
+        context.getExecutionListener().log(Constants.VERBOSE_LEVEL, logprompt + options);
+
+        int result = -1;
+        try {
+            final OverthereConnection connection = new CifsConnectionBuilder(CifsConnectionBuilder.CIFS_PROTOCOL, options,
+                    new DefaultAddressPortMapper()).connect();
+
+            try {
+                OverthereFile dst;
+                if (destination != null && !destination.isEmpty()) {
+                    dst = connection.getFile(destination);
+                    context.getExecutionListener().log(Constants.VERBOSE_LEVEL, logprompt + "Destination file acquired " + dst.getPath());
+                } else {
+                    dst = connection.getTempFile("RD_WINRM_TMP");
+                    context.getExecutionListener().log(Constants.VERBOSE_LEVEL, logprompt + "Temporary file acquired " + dst.getPath());
+                }
+                context.getExecutionListener().log(Constants.VERBOSE_LEVEL, logprompt + "Destination file acquired " + dst.getPath());
+
+                try {
+                    OutputStream w = dst.getOutputStream();
```

Comment:
There is a dedicated method in IOUtils for that: `copy(InputStream input, OutputStream output)`